src/SourceBuild/content/eng/pipelines/ci.yml

```diff
@@ -1,31 +1,27 @@
-# This is the main build definition (PR+CI) for dotnet/dotnet
+# This is the dotnet/dotnet pipeline that is triggered every weekday at midnight (00:00 UTC) for "main" and when changes are pushed to release/* and internal/release/* branches.
 
-trigger:
-  batch: true
+schedules:
+- cron: '0 0 * * Mon-Fri'
+  displayName: Weekday midnight build
   branches:
     include:
     - main
-    - release/*
-    - internal/release/*
-    exclude:
-    - release/*.0.2xx
-    - release/*.0.3xx
-    - release/*.0.4xx
-    - internal/release/*.0.2xx
-    - internal/release/*.0.3xx
-    - internal/release/*.0.4xx
+  batch: true
 
-pr:
+trigger:
+  batch: true
   branches:
-    include:
-    - main
-    - release/*
-    - internal/release/*
+   include:
+   - release/*
+   - internal/release/*
+
+pr: none
 
 stages:
-- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+- ${{ if ne(variables['Build.CronSchedule.DisplayName'], 'Weekday midnight build') }}:

```

Comment:
Would `eq(variables['Build.Reason'], 'Schedule')` make this more robust in case the scheduled time is ever changed?

Reply:
Changed the reason to schedule in https://github.com/dotnet/installer/pull/17370/commits/bf697aa5cab242f80871e0db905b3d85ba24dedf